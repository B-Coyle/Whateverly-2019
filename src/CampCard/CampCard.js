import React, {Component} from 'react';
import './CampCard.scss';
import CampingFilter from '../CampingFilter/CampingFilter.js';
import App from '../App/App.js';
import CampPopUpCard from '../CampPopUpCard/CampPopUpCard.js';

export default class CampCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      favorite: false,
      visited: false,
      popUp: false
    }
  }

  createCampPopUp = () => {
    this.setState({
      popUp: true
    })
  }

  handleDeleteCampCard = (e) => {
    e.preventDefault();
    e.target.closest('section').remove(); 
  }

  handleFavoriteCamp = (event) => {
    event.preventDefault();
    this.setState({
      favorite: !this.state.favorite
    });
  }

  handleVisitedCamp = (event) => {
    event.preventDefault();
    this.setState({
      visited: !this.state.visited
    })
  }

  componentDidUpdate = () => {
    localStorage.setItem(this.props.filteredCampgrounds.name, JSON.stringify(this.state));

  }

  componentDidMount = () => {
    this.setState({
      favorite: JSON.parse(localStorage.getItem(this.props.filteredCampgrounds.name)) || false,
      visited: JSON.parse(localStorage.getItem(this.props.filteredCampgrounds.name)) || false,
    })
  }

  render (props) {
    let campPopUp;
    if(this.state.popUp === true) {
      campPopUp = <CampPopUpCard 
      allTrails = {this.props.allTrails}
      assignedUserCampInput = {this.props.userCampInputValue} />
    }

    return (
      <section className='campingCard'>
        <div>
          <h2 className='campName'>{ this.props.filteredCampgrounds.name }</h2>
          <button className='favoriteButton' onClick={this.handleFavoriteCamp}>  <i className="fas fa-heart"></i></button>
          <button className='visitedButton' onClick={this.handleVisitedCamp}> <i className="fas fa-check"></i></button>
          <button className='deleteButton' onClick={this.handleDeleteCampCard}><i className="far fa-trash-alt"></i></button>
        </div>
        <h3><span className='textLabel'>Location: </span>{ this.props.filteredCampgrounds.location }</h3>
        <h3><span className='textLabel'>Available Seasons: </span>{ this.props.filteredCampgrounds.season }</h3>
        <ul className='campInfo'>
          <li><span className='textLabel'>Elevation: </span>{this.props.filteredCampgrounds.elevation}</li>
          <li><span className='textLabel'>Camping Type: </span>{this.props.filteredCampgrounds.camping}</li>
          <li><span className='textLabel'>Nearby Activities: </span>{this.props.filteredCampgrounds.activities}</li>
        </ul>
        <input className="moreBTn" type="button"value="More" onClick={this.createCampPopUp} />
        {campPopUp}
    </section>
    )
  }
}
