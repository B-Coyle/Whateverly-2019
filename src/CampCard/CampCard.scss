.campingCard{
    display: inline-grid;
    border: #3e272f solid 4px;
    font-size: 1em;
    color: #3e272f;
    background-color: rgba(349, 250, 255, 0.90);
    grid-column-start: 2;
    margin: 3% 18%;
    padding: 5px;
    max-height: 300px;
    width: 70%;
}

.campingCard h3 {
    line-height: 15px;
    font-weight: 600;
}

.campName {
    border-bottom: 2px solid #3e272f;
    font-weight: 500;
}

.favoriteButton,
.visitedButton,
.deleteButton {
    display: inline;
    margin: 0 10%;
    height: 35px;
    width: 13%;
}

.favoriteButton:hover,
.visitedButton:hover,
.deleteButton:hover,
.moreBtn:hover {
    color: white;
    background-color: #af5700; 
}
