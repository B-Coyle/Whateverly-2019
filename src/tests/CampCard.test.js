import React from 'react';
import ReactDOM from 'react-dom';
import CampCard from '../CampCard/CampCard.js';
import { shallow } from 'enzyme';

const mockCamping = [
    {
    "name": "Glacier Basin Campground",
    "location": "Estes Park",
    "season": "May-Sept",
    "elevation": 8500,
    "camping": ["Tent", "Car", "RV"],
    "activities": ["Hiking", "Fishing", "Horseback Riding", "Wildlife"],
    "reservation-available": true,
    "nearest-city": null
      },
      {
    "name": "Grand Junction KOA",
    "location": "Grand Junction",
    "season": "All year",
    "elevation": 4600,
    "camping": ["Tent", "Car", "RV"],
    "activities": ["ATVing"],
    "reservation-available": true,
    "nearest-city": null
    }
]

describe('CampCard', () => {
let wrapper

beforeEach(() => {
    wrapper = shallow(
        <CampCard filteredCampgrounds={mockCamping} />
    )
});

// it('should mock the snapshot', () => {
//     expect(wrapper).toMatchSnapshot({mockCamping});
// });

it('should have a default state', () => {
    expect(wrapper.state()).toEqual({
        favorite: false,
        visited: false,
        popUp: false
    })
});

})