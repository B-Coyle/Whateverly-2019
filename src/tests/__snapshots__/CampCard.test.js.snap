// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampCard should mock the snapshot 1`] = `undefined`;

exports[`Card should mock the snapshot 1`] = `undefined`;
